libsolidity/ast/Types.cpp

```diff
@@ -479,17 +518,35 @@ TypePointer AddressType::binaryOperatorResult(Token::Value _operator, TypePointe
 	return Type::commonType(shared_from_this(), _other);
 }
 
+bool AddressType::operator==(Type const& _other) const
+{
+	if (_other.category() != category())
+		return false;
+	AddressType const& other = dynamic_cast<AddressType const&>(_other);
+	return other.m_stateMutability == m_stateMutability;
+}
+
+TypePointer AddressType::copyForMutabilityIfAddress(StateMutability _stateMutability, TypePointer const& _type)
+{
+	if (auto const* addressType = dynamic_cast<AddressType const*>(_type.get()))
```

Comment:
It might be easier to understand to not use this function, since it can be replaced by
`if (type->category() == Type::Address) type = make_shared<Address>(StateMutability::Payable);`

Reply:
The idea was to use this function in case we add more attributes to the ``address`` type in the future that might need to be copied here - I can remove it for now, though.